src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -38,135 +188,135 @@ export const FileUpload:React.FC<IFileUploadProps>=({
     failureMessage,
     disabled,
 }):React.ReactElement=>{
-    const [height,setHeight]=useState<number|undefined>(undefined)
-    const [totalHeight,setTotalHeight]=useState(0)
-    const [totalHeightPlus,setTotalHeightPlus]=useState(0)
-    const [padding]=useState(10)
-    const [maxHeight,setMaxHeight]=useState<number|undefined>(undefined)
-    const [positionLoading,setPositionLoading]=useState(!isUploading)
-    const [positionTopLoading,setPositionTopLoading]=useState(0)
-    const [opacityLoading,setOpacityLoading]=useState(0)
-    const [positionIsSuccess,setPositionIsSuccess]=useState(!isSuccess)
-    const [opacityIsSuccess,setOpacityIsSuccess]=useState(0)
-    const [loadingContainerHeight,setLoadingContainerHeight]=useState(0)
-    const [widthComponent,setWidthComponent]=useState(0)
-    const [margin]=useState(10)
-    const [widthIsSuccess,setWidthIsSuccess]=useState(0)
-    const [positionIsFailure,setPositionIsFailure]=useState(!isFailure)
-    const [opacityIsFailure,setOpacityIsFailure]=useState(0)
+    const initState:IState={
+        height:undefined,
+        totalHeight:0,
+        totalHeightPlus:0,
+        maxHeight:undefined,
+        padding:10,
+        margin:10,
+        loading:{position:!isUploading,opacity:0},
+        isSuccess:{position:!isSuccess,opacity:0},
+        isFailure:{position:!isFailure,opacity:0},
+        positionTopLoading:0,
+        loadingContainerHeight:0,
+        componentWidth:0,
+        isSuccessWidth:0,
+        isDragEnter:false,
+    }
+    const [state,dispatch]=useReducer(reducer,initState)
 
     // this sets height of the component, is used to transition between heights. 
     useEffect(()=>{
         if(isUploading||isSuccess||isFailure){
-            if(totalHeightPlus){
-                setHeight(totalHeightPlus)
+            if(state.totalHeightPlus){
+                dispatch({type:SET_HEIGHT,value:state.totalHeightPlus})
             }else{
-                setHeight(undefined)
-                setMaxHeight(600)
+                dispatch({type:SET_HEIGHT,value:undefined})
+                dispatch({type:SET_MAXHEIGHT,value:600})
             }
-        }else if(totalHeight){
-            setHeight(totalHeight)   
+        }else if(state.totalHeight){
+            dispatch({type:SET_HEIGHT,value:state.totalHeight})
         }
-    },[isUploading,isSuccess,isFailure,totalHeight,totalHeightPlus])
+    },[isUploading,isSuccess,isFailure,state.totalHeight,state.totalHeightPlus])
 
     // this is to calculate (set) some values
     useEffect(()=>{
         if(containerRef.current?.scrollHeight){
             const contentHeight=containerRef.current.scrollHeight
-            setTotalHeight(contentHeight-padding*2)
-            setMaxHeight(contentHeight-padding*2)
+            dispatch({type:SET_TOTALHEIGHT,value:contentHeight-state.padding*2})
+            dispatch({type:SET_MAXHEIGHT,value:contentHeight-state.padding*2})
```

Comment:
These can be combined into one dispatch, and the same for line 215.

Reply:
how do you combine it into only one dispatch?

Reply:
yes, I know it. I will do it...